src/dvr/dvr_rec.c

```diff
@@ -124,8 +128,21 @@ dvr_rec_subscribe(dvr_entry_t *de)
     }
   }
 
-  if(stat(de->de_config->dvr_storage, &st) || !S_ISDIR(st.st_mode)) {
-    tvherror(LS_DVR, "the directory '%s' is not accessible", de->de_config->dvr_storage);
+  errSaved = 0;
+  statRet = stat(de->de_config->dvr_storage, &st);
+  errSaved = errno;
+
+  //If the stat() failed, show the ERRNO.
+  if(statRet != 0) {
+    strerror_r(errSaved, errText, sizeof(errText));
```

Comment:
Use `strerror` so you don't need those 1024 byte buffer.

Reply:
> Use `strerror` so you don't need those 1024 byte buffer.

True, however, `strerror` appears not to be thread-safe.  This is why I used `strerror_r` instead.

https://man7.org/linux/man-pages/man3/strerror.3.html

Do you still want to use `strerror`?

Reply:
It's not guaranteed to be thread-safe, it actually depends on the implementation. The "worst" thing that can happen is, that 2 error messages are logged at the same time from multiple threads and one message is not shown and the other one is shown twice (so strerror returns the wrong string). It's pretty unlikely to have a scenario where 2 errors are logged at the same time, so that's fine.

Also some implementations return a pointer while holding all the strings in memory, in that case there is no issue with regards to concurrency.

Reply:
I will use `strerror` as required.